lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -150,13 +157,28 @@ static bool validateCodingKeysEnum(DerivedConformance &derived,
   for (auto *varDecl : derived.Nominal->getStoredProperties()) {
     if (!varDecl->isUserAccessible())
       continue;
-
-    properties[getVarNameForCoding(varDecl)] = varDecl;
+    // When we have invalid redeclaration we may endup with duplicated
+    // members here.
+    auto varIdentifier = getVarNameForCoding(varDecl);
+    if (properties.count(varIdentifier) != 0)
```

Comment:
Nit: 
```suggestion
    if (properties.count(varIdentifier))
```

and below.